server/src/test/java/com/microsoft/java/bs/core/internal/server/IntegrationTest.java

```diff
@@ -0,0 +1,431 @@
+    }
+
+    @Override
+    public void onBuildLogMessage(LogMessageParams params) {
+      logMessages.add(params);
+      synchronized (this) {
+        notify();
+      }
+    }
+
+    @Override
+    public void onBuildTaskStart(TaskStartParams params) {
+      if (params.getDataKind() != null) {
+        if (params.getDataKind().equals("compile-task")) {
+          compileTasks.add(JsonUtils.toModel(params.getData(), CompileTask.class));
+        } else if (params.getDataKind().equals("test-start")) {
+          testStarts.add(JsonUtils.toModel(params.getData(), TestStartEx.class));
+        } else {
+          fail("Task Start kind not handled " + params.getDataKind());
+        }
+      }
+      startReports.add(params);
+      synchronized (this) {
+        notify();
+      }
+    }
+
+    @Override
+    public void onBuildTaskProgress(TaskProgressParams params) {
+      // do nothing
+    }
+
+    @Override
+    public void onBuildTaskFinish(TaskFinishParams params) {
+      if (params.getDataKind() != null) {
+        if (params.getDataKind().equals("compile-report")) {
+          compileReports.add(JsonUtils.toModel(params.getData(), CompileReport.class));
+        } else if (params.getDataKind().equals("test-report")) {
+          testReports.add(JsonUtils.toModel(params.getData(), TestReport.class));
+        } else if (params.getDataKind().equals("test-finish")) {
+          testFinishes.add(JsonUtils.toModel(params.getData(), TestFinishEx.class));
+        } else {
+          fail("Task Finish kind not handled " + params.getDataKind());
+        }
+      }
+      finishReports.add(params);
+      synchronized (this) {
+        notify();
+      }
+    }
+
+    @Override
+    public void onBuildPublishDiagnostics(PublishDiagnosticsParams params) {
+      // do nothing
+    }
+
+    @Override
+    public void onBuildTargetDidChange(DidChangeBuildTarget params) {
+      // do nothing
+    }
+  }
+
+  @BeforeAll
+  static void beforeClass() {
+    String pluginDir = Paths.get(System.getProperty("user.dir"),
+        "build", "libs", "plugins").toString();
+    System.setProperty(Launcher.PROP_PLUGIN_DIR, pluginDir);
+    System.setProperty("bsp.plugin.reloadworkspace.disabled", "true");
+  }
+
+  @AfterAll
+  static void afterClass() {
+    System.clearProperty(Launcher.PROP_PLUGIN_DIR);
+    System.clearProperty("bsp.plugin.reloadworkspace.disabled");
+  }
+
+  protected static InitializeBuildParams getInitializeBuildParams(String projectDir) {
+    File root = Paths.get(
+        System.getProperty("user.dir"),
+        "..",
+        "testProjects",
+        projectDir).toFile();
+
+    BuildClientCapabilities capabilities =
+        new BuildClientCapabilities(SupportedLanguages.allBspNames);
+    return new InitializeBuildParams(
+        "test-client",
+        "0.1.0",
+        "0.1.0",
+        root.toURI().toString(),
+        capabilities);
+  }
+
+  protected static InitializeBuildParams getInitializedBuildParamsWithJdks(
```

Comment:
Put this as a member of LifecycleServiceIntegrationTest?

Reply:
Fixed in https://github.com/microsoft/build-server-for-gradle/pull/165/commits/7e4749a704b01421c5b6282ea0204d8856b32636